src/model/validation.rs

```diff
@@ -0,0 +1,242 @@
+//! Logic for model ([`Manifest`]) validation
+//!
+
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::OnceLock;
+
+use anyhow::{Context as _, Result};
+use serde::{Deserialize, Serialize};
+
+use crate::model::{LinkProperty, Manifest, TraitProperty};
+
+type NamespaceName = String;
+type PackageName = String;
+type InterfaceName = String;
+
+/// A namespace -> package -> interface lookup
+type KnownInterfaceLookup =
+    HashMap<NamespaceName, HashMap<PackageName, HashMap<InterfaceName, ()>>>;
+
+/// Hard-coded list of known namespaces/packages and the interfaces they contain.
+///
+/// Using an interface that is *not* on this list is not an error --
+/// custom interfaces are expected to not be on this list, but when using
+/// a known namespace and package, interfaces should generally be well known.
+static KNOWN_INTERFACE_LOOKUP: OnceLock<KnownInterfaceLookup> = OnceLock::new();
+
+/// Get the static list of known interfaces
+fn get_known_interface_lookup() -> &'static KnownInterfaceLookup {
+    KNOWN_INTERFACE_LOOKUP.get_or_init(|| {
+        HashMap::from([
+            (
+                "wrpc".into(),
+                HashMap::from([
+                    ("keyvalue".into(), HashMap::from([("atomics".into(), ())])),
+                    ("blobstore".into(), HashMap::from([("atomics".into(), ())])),
+                    ("config".into(), HashMap::from([("runtime".into(), ())])),
+                ]),
+            ),
+            (
+                "wasi".into(),
+                HashMap::from([
+                    ("keyvalue".into(), HashMap::from([("atomics".into(), ())])),
+                    ("blobstore".into(), HashMap::from([("atomics".into(), ())])),
```

Comment:
Same comment re: blobstore/atomics

Reply:
See [this comment](https://github.com/wasmCloud/wadm/pull/274#discussion_r1591390096) -- the list in the PR is nowhere near complete but I was more looking for any design feedback!